sdks/java/extensions/sql/zetasql/src/main/java/org/apache/beam/sdk/extensions/sql/zetasql/ZetaSqlCalciteTranslationUtils.java

```diff
@@ -180,12 +182,19 @@ private static RelDataType toCalciteStructType(
 
   private static List<String> getFieldNameList(List<StructField> fields) {
     ImmutableList.Builder<String> b = ImmutableList.builder();
+    Set<String> usedName = new HashSet<>();
     for (int i = 0; i < fields.size(); i++) {
       String name = fields.get(i).getName();
-      if ("".equals(name)) {
-        name = "$col" + i; // avoid empty field names because Beam does not allow duplicate names
+      // Follow the same way that BigQuery handles unspecified or duplicate field name
+      if ("".equals(name) || usedName.contains(name)) {
```

Comment:
This does not actually match the behavior of BigQuery. To match BigQuery you should remove the `HashSet` and instead replace anything starting with `_field_` with the correct index for that column.

(In BigQuery, this condition would be `"".equals(name) || name.startsWith("_field_")`)

Reply:
I run `SELECT STRUCT("abc" as a, 3 as a)` on BigQuery. The result is a row of 2 fields whose names are `f0_.a` and `f0_._field_2`

Reply:
I run `SELECT STRUCT("abc" AS _field_22, 3 AS _field_23)` on BigQuery. The result is a row of 2 files whoes names are `f0_._field_1` and `f0_._field_2`.

Reply:
OK I got your point now. So I believe the condition should be `"".equals(name) || name.startsWith("_field_") || usedName.contains(name)`. This way we can avoid the exception.